### PR TITLE
check-alsabat.sh: dump amixer settings always

### PR DESCRIPTION
**Pull request by fredoh9**
amixer settings are always required. good case for reference and bad case for more debugging.

### CODE REVIEW

**Comment by fredoh9** [comment]
related discussion: https://github.com/thesofproject/sof/issues/9018

**Review by marc-hb — APPROVED** [review]

File: test-case/check-alsabat.sh [hunk 1]
```diff
@@ -121,14 +121,16 @@ alsabat -P$pcm_p --standalone -n $frames -c $channel_p -r $rate -f $format -F $f
 # playback may have low latency, add one second delay to aviod recording zero at beginning.
 sleep 1
 
+# dump amixer contents always.
+# Good case amixer settings is for reference, bad case for debugging.
+amixer contents > "$LOG_ROOT"/amixer_settings.txt
```
Comment (plbossart) [inline test-case/check-alsabat.sh:126]: this is not good enough @fredoh9, you need to get the card number or you will get a limited set of controls

````
pbossart@pbossart-mobl6:/data/pbossart/ktest/work$ amixer controls
numid=4,iface=MIXER,name='Master Playback Switch'
numid=3,iface=MIXER,name='Master Playback Volume'
numid=2,iface=MIXER,name='Capture Switch'
numid=1,iface=MIXER,name='Capture Volume'
pbossart@pbossart-mobl6:/data/pbossart/ktest/work$ amixer -Dhw:0 controls
numid=11,iface=CARD,name='HDMI/DP,pcm=3 Jack'
numid=17,iface=CARD,name='HDMI/DP,pcm=4 Jack'
numid=23,iface=CARD,name='HDMI/DP,pcm=5 Jack'
numid=10,iface=CARD,name='Headphone Jack'
numid=9,iface=CARD,name='Mic Jack'
numid=2,iface=MIXER,name='Master Playback Switch'
numid=1,iface=MIXER,name='Master Playback Volume'
numid=8,iface=MIXER,name='Mic Boost Volume'
numid=5,iface=MIXER,name='Mic Playback Switch'
numid=4,iface=MIXER,name='Mic Playback Volume'
numid=7,iface=MIXER,name='Capture Switch'
numid=6,iface=MIXER,name='Capture Volume'
numid=3,iface=MIXER,name='Loopback Mixing'
numid=12,iface=MIXER,name='IEC958 Playback Con Mask'
numid=18,iface=MIXER,name='IEC958 Playback Con Mask',index=1
numid=24,iface=MIXER,name='IEC958 Playback Con Mask',index=2
numid=13,iface=MIXER,name='IEC958 Playback Pro Mask'
numid=19,iface=MIXER,name='IEC958 Playback Pro Mask',index=1
numid=25,iface=MIXER,name='IEC958 Playback Pro Mask',index=2
numid=14,iface=MIXER,name='IEC958 Playback Default'
numid=20,iface=MIXER,name='IEC958 Playback Default',index=1
numid=26,iface=MIXER,name='IEC958 Playback Default',index=2
numid=15,iface=MIXER,name='IEC958 Playback Switch'
numid=21,iface=MIXER,name='IEC958 Playback Switch',index=1
numid=27,iface=MIXER,name='IEC958 Playback Switch',index=2
numid=34,iface=MIXER,name='EQIIR2.0 eqiir_bytes_2'
numid=32,iface=MIXER,name='PGA1.0 1 Master Playback Volume'
numid=33,iface=MIXER,name='PGA2.0 2 Master Capture Volume'
numid=38,iface=MIXER,name='PGA30.0 30 Playback Volume'
numid=39,iface=MIXER,name='PGA31.0 31 Playback Volume'
numid=35,iface=MIXER,name='PGA7.0 7 Master Playback Volume'
numid=36,iface=MIXER,name='PGA8.0 8 Master Playback Volume'
numid=37,iface=MIXER,name='PGA9.0 9 Master Playback Volume'
numid=16,iface=PCM,name='ELD',device=3
numid=29,iface=PCM,name='Playback Channel Map',device=3
numid=22,iface=PCM,name='ELD',device=4
numid=30,iface=PCM,name='Playback Channel Map',device=4
numid=28,iface=PCM,name='ELD',device=5
numid=31,iface=PCM,name='Playback Channel Map',device=5
````

Comment (plbossart) [reply test-case/check-alsabat.sh:126]: you could filter the ELD/HDMI/IEC stuff btw since it's irrelevant for alsabat.

**Review by marc-hb — COMMENTED** [review]
That command was already "wrong" in the first place, this PR is merely moving it and making that more obvious.

I think we can merge this one-line move already, it will give us a lot more sample data which will make it more obvious what's missing and spurious.

**Pull request merged**